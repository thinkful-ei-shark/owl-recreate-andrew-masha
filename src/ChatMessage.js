import React from 'react';

function ChatMessage(props) {
  const time = new Date(props.event.timestamp);
  return (
    <li className="chat-message">
      <img src={props.participant.avatar} alt="avatar" />,
      <div className="byline">
        <span className="participant-name">{props.participant.name}</span>
        <span className="eventTime">{time.toDateString()}</span>
      </div>,
      <div className="message-text">{props.event.message}</div>
    </li>
  )
}

export default ChatMessage;