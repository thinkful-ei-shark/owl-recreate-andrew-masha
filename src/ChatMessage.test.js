import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import ChatMessage from './ChatMessage.js';

const testProps = {
  event: {
    participantId: 3,
    type: 'message',
    message: 'Hello world',
    time: 1548852646559,
    timestamp: 1548852484247
  },
  participant: {

    id: 1,
    name: 'Koren Templeton',
    avatar:
      'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
    inSession: true,
    onStage: true

  }
}

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <ChatMessage event={testProps.event} participant={testProps.participant} />,
    div
  );

  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(<ChatMessage
      event={testProps.event}
      participant={testProps.participant} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})