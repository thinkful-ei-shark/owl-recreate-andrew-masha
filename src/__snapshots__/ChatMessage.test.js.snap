// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<li
  className="chat-message"
>
  <img
    alt="avatar"
    src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
  />
  ,
  <div
    className="byline"
  >
    <span
      className="participant-name"
    >
      Koren Templeton
    </span>
    <span
      className="eventTime"
    >
      Wed Jan 30 2019
    </span>
  </div>
  ,
  <div
    className="message-text"
  >
    Hello world
  </div>
</li>
`;
